src/spyglass/decoding/v1/clusterless.py

```diff
@@ -303,6 +302,35 @@ def load_environments(key):
 
         return classifier.environments
 
+    @staticmethod
+    def _get_interval_range(key):
+        encoding_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["encoding_interval"],
+            }
+        ).fetch1("valid_times")
+
+        decoding_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["decoding_interval"],
+            }
+        ).fetch1("valid_times")
+
+        return (
+            min(
+                np.asarray(encoding_interval).min(),
+                np.asarray(decoding_interval).min(),
+            ),
+            max(
+                np.asarray(encoding_interval).max(),
+                np.asarray(decoding_interval).max(),
+            ),
+        )
```

Comment:
I think you'd get negligible performance increases by concatenating first
```suggestion
        both_intervals = np.concatenate((encoding_interval, decoding_interval))
        
        return both_intervals.min(), both_intervals.max()
```
I see that the same logic is repeated below on `SortedSpikesDecoding`. That suggests to me that this is a helper method for `decoding/utils/` or `decoding/vX/utils`